NuRadioReco/framework/base_trace.py

```diff
@@ -69,32 +70,65 @@ def get_frequency_spectrum(self):
             self.__time_domain_up_to_date = False
         return np.copy(self._frequency_spectrum)
 
-    def set_trace(self, trace, sampling_rate):
+    def set_trace(self, trace, sampling_rate=None):
         """
-        sets the time trace
+        Sets the time trace
 
         Parameters
         ----------
-        trace: np.array of floats
-            the time series
-        sampling_rate: float
-            the sampling rage of the trace, i.e., the inverse of the bin width
+        trace : np.array of floats
+            The time series
+        sampling_rate : float or str (Default: None)
+            The sampling rate of the trace, i.e., the inverse of the bin width.
+            If `sampling_rate="same"`, sampling rate is not changed (requires previous initialisation).
+            If `sampling_rate=None` raise an error.
         """
         if trace is not None:
             if trace.shape[trace.ndim - 1] % 2 != 0:
-                raise ValueError(('Attempted to set trace with an uneven number ({}) of samples. '
-                                 'Only traces with an even number of samples are allowed.').format(trace.shape[trace.ndim - 1]))
+                raise ValueError(f'Attempted to set trace with an uneven number ({trace.shape[trace.ndim - 1]}) '
+                                 'of samples. Only traces with an even number of samples are allowed.')
         self.__time_domain_up_to_date = True
         self._time_trace = np.copy(trace)
-        self._sampling_rate = sampling_rate
+
         self._frequency_spectrum = None
 
-    def set_frequency_spectrum(self, frequency_spectrum, sampling_rate):
+        if isinstance(sampling_rate, str) and sampling_rate.lower() == "same":
+            if self._sampling_rate is None:
+                raise ValueError("You specified to keep the sampling rate "
+                                 "but no value have been set previously.")
+                pass  # keep value of self._sampling_rate
+        elif sampling_rate is not None:
+            self._sampling_rate = sampling_rate
+        else:
+            raise ValueError("You have to specify a sampling rate for `BaseTrace.set_trace(...)`")
+
+    def set_frequency_spectrum(self, frequency_spectrum, sampling_rate="same"):
```

Comment:
Here the docstring says the default value for sampling_rate is None. Same as above, if the default argument raises an error, why not just have no default value?

Reply:
done